### PR TITLE
feat: Account form error handling

### PR DESCRIPTION
**Pull request by Yserz**
image

### CODE REVIEW

File: app/script/auth/module/action/ValidationError.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+export default class ValidationError extends Error {
+  constructor(params) {
+    super();
```
Comment (AndyLnd) [inline app/script/auth/module/action/ValidationError.js]: `super(params)`

File: app/script/auth/module/action/ValidationError.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    this.name = this.constructor.name;
+    this.label = params.label;
+    this.message = params.label;
+  }
+
+  is = label => {
```
Comment (bennycode) [inline app/script/auth/module/action/ValidationError.js]: No `static`, no `const`, what is `is`?

File: app/script/auth/module/action/ValidationError.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  static handleValidationState(fieldName, validationState) {
+    const field = ValidationError.getFieldByName(fieldName);
+    const validationStateKeys = ValidationError.getAllPropertyNames(validationState);
+    const errorKeys = [];
+    Object.keys(ValidationError.ERROR).forEach(errorName => errorKeys.push(ValidationError.ERROR[errorName]));
+    for (const key of validationStateKeys) {
+      if (errorKeys.includes(key) && validationState[key]) {
+        return new ValidationError({
+          label: field[ValidationError.getErrorByKey(key)],
+        });
```
Comment (gregor) [inline app/script/auth/module/action/ValidationError.js]: This looks like you can just to a `map` on `Object.keys` or with the help of babel even on `Object.entries()` to full `errorKeys` immeditately without all these validations.

File: app/script/auth/module/action/ValidationError.js (hunk 2 above)
Comment (Yserz) [reply app/script/auth/module/action/ValidationError.js]: It's a convenience method to check if an instance of an error has a given label.
`if(someValidationError.is(ValidationError.NAME.PATTERN_MISMATCH)) ...`

**Pull request merged**